### ISSUE TITLE
Proguard configuration

### ISSUE THREAD

**Issue by Gaket** [google/protobuf-gradle-plugin#217]
We use a Proguard in our app. What is the Proguard config needed to work with Protobuf plugin? Next config works for my test project:
```
-dontwarn sun.misc.Unsafe
# Keep model classes
-keep class org.test.tutorial.** { *; }
```

Is there a way not to keep the model classes from obfuscation?

**Comment by zhangkun83** [comment]
Why do you need to special-case the generated classes? They are compiled in the same way other files are.

**Comment by Gaket** [comment]
Oh, I see. I used this project as a test project: https://github.com/Gaket/demo_android_protobuf_simple

And in that project the initial author used `MyObjectClass.toString()` call to show the incoming data. And this call returned printed obfuscated nonsense.

Only after your comment I checked implementation of generated `toString()` and saw that it is reflection-based (and you even described it in and the reasoning for it the java doc, my bad). I've checked the actual object values and they look good.

I checked another Protobuf implementation, Wire (https://github.com/square/wire) and they contain these Proguard rules. Applying them to the project fixed the issue I had had but now I see that it was not needed actually, thank you.
